src/Raven.Server/Documents/Indexes/Static/IndexCompilationCache.cs

```diff
@@ -17,7 +17,7 @@ public class IndexCompilationCache
         /// will build up and tear down a server frequently, so we can still reduce the cost of compiling 
         /// the indexes.
         /// </summary>
-        private readonly ConcurrentDictionary<IndexCacheKey, Lazy<StaticIndexBase>> _staticIndexCache = new ConcurrentDictionary<IndexCacheKey, Lazy<StaticIndexBase>>();
+        private static readonly ConcurrentDictionary<IndexCacheKey, Lazy<StaticIndexBase>> _staticIndexCache = new ConcurrentDictionary<IndexCacheKey, Lazy<StaticIndexBase>>();
```

Comment:
But now we are back at making this static globally, no? So what is the purpose?

Reply:
This one is for C# indexes. Second one is for JS ones. I'm checking all usages of cache now, maybe we can remove JS ones from cache entirely

Reply:
The main reason we have a cache for C# is to control the # of assemblies.
For JS, we probably can drop it

Reply:
Done. Thx!